sleet-api-service/src/main/java/com/sleet/api/service/QuoteService.java

```diff
@@ -30,15 +31,45 @@ public QuoteService(final String apiKey) {
      * @param ticker to get quote info for
      * @return an {@link Equity} with quote information
      */
-    public Equity getQuote(final String ticker) {
+    public Optional<Equity> getQuote(final String ticker) {
 
         final String url = API_URL + ticker + QUOTE_URL;
         try {
             final ResponseEntity<JsonNode> json = restTemplate.getForEntity(url, JsonNode.class);
-            return mapper.readValue(json.getBody().get(ticker).toString(), Equity.class);
-        } catch(Exception e) {
-            LOG.error("Could not retrieve quote info for " + ticker, e);
+            Equity equity = mapper.readValue(Objects.requireNonNull(json.getBody())
+                    .get(ticker).toString(), Equity.class);
+            return Optional.ofNullable(equity);
+        } catch (Exception e) {
+            LOG.error("Could not retrieve quote for " + ticker, e);
```

Comment:
As part of my rework of OptionService, I decided it may be better to just throw the exception back to the framework calling these methods, and let it decide how to handle bad state rather then logging and returning null for the most part. We can always improve the specificity of the exceptions later on if required (timeout, deserialization, etc). Thoughts?

Reply:
That makes sense.  In practice, I have also seen some service-not-available type of issues originating from TDA.

The code would also be a little cleaner because you can potentially use the `orElse(...)` syntax in some cases.